File: src/jvis/data/stacks/nodejs-express/files/src/application/dto/item.dto.ts
import { z } from 'zod';

export const CreateItemSchema = z.object({
  name: z.string().min(1).max(255),
  description: z.string().max(1000).nullable().optional(),
});

export const UpdateItemSchema = z.object({
  name: z.string().min(1).max(255).optional(),
  description: z.string().max(1000).nullable().optional(),
});

export type CreateItemDto = z.infer<typeof CreateItemSchema>;
export type UpdateItemDto = z.infer<typeof UpdateItemSchema>;

File: src/jvis/data/stacks/rust-axum/files/src/application/dto/item_dto.rs
use serde::Deserialize;

#[derive(Debug, Deserialize)]
pub struct CreateItemDto {
    pub name: String,
    pub description: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateItemDto {
    pub name: Option<String>,
    pub description: Option<String>,
}
